File: backend/dalathasfarm/src/main/java/com/example/dalathasfarm/responses/order/OrderResponse.java
package com.example.dalathasfarm.responses.order;

import com.example.dalathasfarm.models.Order;
import com.example.dalathasfarm.responses.orderdetail.OrderDetailResponse;
import jakarta.validation.constraints.NotNull;
import lombok.*;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.util.List;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class OrderResponse {
    private Integer id;
    private Integer userId;
    private String fullName;
    private String email;
    private String phoneNumber;
    private String address;
    private String note;
    private String status;
    private LocalDate orderDate;
    private LocalDate shippingDate;
    private BigDecimal totalMoney;
    private String paymentMethod;
    private String shippingMethod;
    private String platform;
    private Boolean isActive;
    private String vnpTxnRef;
    private String invoiceFile;
    private List<OrderDetailResponse> orderDetailResponses;

    public static OrderResponse fromOrder(Order order, List<OrderDetailResponse> orderDetailResponses) {
        return OrderResponse.builder()
                .id(order.getId())
                .userId(order.getUser().getId())
                .fullName(order.getFullName())
                .email(order.getEmail())
                .phoneNumber(order.getPhoneNumber())
                .address(order.getAddress())
                .note(order.getNote())
                .status(order.getStatus().name())
                .orderDate(order.getOrderDate())
                .shippingDate(order.getShippingDate())
                .totalMoney(order.getTotalMoney())
                .paymentMethod(order.getPaymentMethod().name())
                .shippingMethod(order.getShippingMethod().name())
                .platform(order.getPlatform().name())
                .isActive(order.getIsActive())
                .vnpTxnRef(order.getVnpTxnRef())
                .invoiceFile(order.getInvoiceFile())
                .orderDetailResponses(orderDetailResponses)
                .build();
    }
}

File: android/lib/responses/order_response.dart
import 'package:android/responses/order_detail_response.dart';

class OrderResponse {
  final int id;
  final String address;
  final int userId;
  final String? note;
  final String? email;
  final double totalMoney;
  final String phoneNumber;
  final DateTime orderDate;
  final String fullName;
  final String status;
  final String paymentMethod;
  final String shippingMethod;
  final String platform;
  final DateTime shippingDate;
  final bool isActive;
  final String? vnpTxnRef;
  final String invoiceFile;
  final List<OrderDetailResponse> orderDetailResponses;

  OrderResponse({
    required this.id,
    required this.address,
    required this.userId,
    this.note,
    this.email,
    required this.totalMoney,
    required this.phoneNumber,
    required this.orderDate,
    required this.fullName,
    required this.status,
    required this.paymentMethod,
    required this.shippingMethod,
    required this.shippingDate,
    required this.isActive,
    required this.platform,
    this.vnpTxnRef,
    required this.invoiceFile,
    required this.orderDetailResponses,
  });

  factory OrderResponse.fromJson(Map<String, dynamic> json) {
    return OrderResponse(
      id: json['id'],
      address: json['address'],
      userId: json['userId'],
      note: json['note'],
      email: json['email'],
      totalMoney: (json['totalMoney'] as num).toDouble(),
      phoneNumber: json['phoneNumber'],
      orderDate: DateTime.parse(json['orderDate']),
      fullName: json['fullName'],
      status: json['status'],
      platform: json['platform'],
      paymentMethod: json['paymentMethod'],
      shippingMethod: json['shippingMethod'],
      shippingDate: DateTime.parse(json['shippingDate']),
      isActive: json['isActive'],
      vnpTxnRef: json['vnpTxnRef'],
      invoiceFile: json['invoiceFile'],
      orderDetailResponses:
          (json['orderDetailResponses'] as List)
              .map((item) => OrderDetailResponse.fromJson(item))
              .toList(),
    );
  }
}
